src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { Container, Button } from '@material-ui/core';
+import TextFieldComp from '../components/TextFieldComp';
+
+function ScholarshipForm() {
+  const [formFieldStates, setFormFieldStates] = useState({
+    name: '',
+    deadline: new Date(),
+    description: '',
+    amountType: '',
+    amount: 0,
+    minAmount: 0,
+    maxAmount: 0,
+    website: '',
+    scholarshipType: '',
+    active: false,
+  });
+
+  function handleFieldChange(name, value) {
+    setFormFieldStates((prev) => ({ ...prev, [name]: value }));
+  }
+
+  return (
+    <Container maxWidth="md">
+      <form>
+        <h1>Submit a Scholarship</h1>
+        <TextFieldComp
+          id="name"
+          label="Scholarship Name *"
+          value={formFieldStates.name}
+          onChange={handleFieldChange}
+        />
```

Comment:
These could be written using `<TextField>` instead:
```jsx
<TextField id="name" label="Scholarship Name *" value={state.name} onChange={updateFn('name')}
```
Where `updateFn` could be
```jsx
function updateFn(id) {
  return (e) => setState({...state, id: e.target.value});
}
```